csrc/quantization/fp8/fp8_cuda_kernels.cu

```diff
@@ -17,6 +18,15 @@ __device__ __forceinline__ float atomicMaxFloat(float* addr, float value) {
     return old;
 }
 
+#define FP8_E4M3_MAX std::numeric_limits<c10::Float8_e4m3fn>::max()
+
+template<typename scalar_t>
+__device__ __forceinline__ c10::Float8_e4m3fn scaled_fp8_conversion(const scalar_t val, const float scale) {
+  float x = static_cast<float>(val) / scale;
+  float r = max(-FP8_E4M3_MAX, min(x, FP8_E4M3_MAX));
+  return static_cast<c10::Float8_e4m3fn>(r);
+}
```

Comment:
Here is my suggestion using intrinsics. I saw the same deterministic output on my smoke test, but it is worth doing another eval in your setup. 
```suggestion
template<typename scalar_t>
__device__ __forceinline__ c10::Float8_e4m3fn scaled_fp8_conversion(const scalar_t val, const float scale) {
  float x = __fdiv_rn(static_cast<float>(val), scale);
  float r = fmaxf(-FP8_E4M3_MAX, fminf(x, FP8_E4M3_MAX));
  return static_cast<c10::Float8_e4m3fn>(r);
}
```

Reply:
Unfortunately this doesn't seem to make it faster. The only thing that is fast so far is
```
template<typename scalar_t>
__device__ __forceinline__ c10::Float8_e4m3fn scaled_fp8_conversion(const scalar_t val, const float scale) {
  float x = static_cast<float>(val) / scale;
  return __nv_cvt_float_to_fp8(x, __NV_SATFINITE, __NV_E4M3);
}
```
but it doesn't seem to be correct, e.g. I'm seeing
```python
from vllm import _custom_ops as ops
import torch

t = 10000 * torch.ones(1, dtype=torch.float16, device="cuda")
ops.scaled_fp8_quant(t, torch.ones(1, device="cuda"))
```
returns
```
(tensor([128.], device='cuda:0', dtype=torch.float8_e4m3fn),
 tensor([1.], device='cuda:0'))
```
but it should be `448.0` (the max of `float8_e4m3`). If you have any ideas what is going wrong here I'd love to know :D

Reply:
This also works but is not super fast either:
```c++
float x = static_cast<float>(input[i]) / *scale;
auto r = __nv_cvt_float_to_fp8(x, __NV_SATFINITE, __NV_E4M3);
out[i] = c10::Float8_e4m3fn(r, c10::Float8_e4m3fn::from_bits());
```

Reply:
I don't think the performance difference I was seeing was actually from this -- I implemented `fmax` and `fmin` now but kept the division since it didn't seem to make a difference for performance @mgoin :)

Reply:
Nice - we should work harder on the load/store overlapping around the core operation